okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/RealConnection.kt

```diff
@@ -364,19 +460,10 @@ class RealConnection(
   }
 
   @Throws(IOException::class)
-  private fun connectTls(connectionSpecSelector: ConnectionSpecSelector) {
+  private fun connectTls(sslSocket: SSLSocket, connectionSpec: ConnectionSpec) {
     val address = route.address
-    val sslSocketFactory = address.sslSocketFactory
     var success = false
-    var sslSocket: SSLSocket? = null
     try {
-      // Create the wrapper over the connected socket.
-      sslSocket = sslSocketFactory!!.createSocket(
-        rawSocket, address.url.host, address.url.port, true /* autoClose */
-      ) as SSLSocket
-
-      // Configure the socket's ciphers, TLS versions, and extensions.
-      val connectionSpec = connectionSpecSelector.configureSecureSocket(sslSocket)
       if (connectionSpec.supportsTlsExtensions) {
```

Comment:
Tangential, should we remove this flag? Is it still useful in 2022, especially as most environments things like SNI are baked in and we don't control them, and don't actively disable them. Looking at configureTlsExtensions

Platform - does nothing.
Jdk9Platform - ALPN
Conscrypt - ALPN + setUseSessionTickets
AndroidSocketAdapter - ALPN + session tickets + SNI
Android10SocketAdapter - ALPN 
ConscryptSocketAdapter - ALPN + setUseSessionTickets

Reply:
I'm unsure and a bit risk averse with the things in Platform. Your call!

Reply:
Let me think about it.  Likely yes.